From: Michael Leibowitz <michael.leibowitz@intel.com>
Date: Fri, 21 Feb 2014 00:49:35 +0000 (-0800)
Subject: Adapt configure.ac to the new pam smack module (PTREL-549)
X-Git-Tag: accepted/tizen/generic/20140225.074541~1
X-Git-Url: https://review.tizen.org/gerrit/gitweb?p=platform%2Fupstream%2Fpam.git;a=commitdiff_plain;h=e9fdd38aa536be354b972b74e4fef8f961e7c73d

Add a pam module to set smack label (PTREL-549)

After some back and forth, this was deemed to be sufficient for now, as
other solutions where wrong in an equal number of cases, but where
substantially more complicated.

Adds a session module that just sets context to "User"

[Patrick Ohly: split out the configure part from the original patch]

Signed-off-by: Michael Leibowitz <michael.leibowitz@intel.com>
---

diff --git a/configure.ac b/configure.ac
index ae762a2..2749f0c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -498,6 +498,9 @@ if test ! -z "$LIBSELINUX" ; then
     LIBS=$BACKUP_LIBS
 fi
 
+dnl hacks for pam_smack
+PKG_CHECK_MODULES([LIBSMACK], [libsmack])
+
 dnl Checks for header files.
 AC_HEADER_DIRENT
 AC_HEADER_STDC
@@ -625,6 +628,7 @@ AC_CONFIG_FILES([Makefile libpam/Makefile libpamc/Makefile libpamc/test/Makefile
 	modules/pam_unix/Makefile modules/pam_userdb/Makefile \
 	modules/pam_warn/Makefile modules/pam_wheel/Makefile \
 	modules/pam_xauth/Makefile doc/Makefile doc/specs/Makefile \
+	modules/pam_smack/Makefile \
 	doc/man/Makefile doc/sag/Makefile doc/adg/Makefile \
 	doc/mwg/Makefile examples/Makefile tests/Makefile \
 	xtests/Makefile])
